src/main/java/com/ithaca/group/GroupService.java

```diff
@@ -0,0 +1,21 @@
+package com.ithaca.group;
+
+/**
+ * Created by David on 4/28/16.
+ */
+public interface GroupService {
+
+    Book_Group create(Long userId, Long bookId);
+
+    Book_Group join(Long userId, Long groupId);
+
+    Book_Group find(Long userId, Long groupId);
+
+    Book_Group post(Long userId, Long groupId, String text);
+
+    Book_Group reply(Long userId, Long groupId, Long postId, String text);
+
+    Book_Group edit(Long userId, Long groupId, Long postId, String text);
+
+    Book_Group upvote(Long userId, Long groupId, Long postId);
```

Comment:
That's the next thing I'm going to try to throw together, with the modifications we talked about with the clients.

How I see it working is every Post has a boolean called "reported" originally set False

Any user has the ability to report a post, which flips the boolean of the Post to True. This does not remove the comment.

If someone else reports the same Post, which would be easy to tell because it will have the reported flag already set to true, then we will delete it. Basically a warning system.